packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -264,46 +264,41 @@ export function readContext<T>(
   context: ReactContext<T>,
   observedBits: void | number | boolean,
 ): T {
-  if (typeof observedBits !== 'number') {
-    if (observedBits === false) {
-      // Do not observe updates
-      observedBits = 0;
+  if (lastContextWithAllBitsObserved === context) {
+    // Nothing to do. We already observe everything in this context.
+  } else if (observedBits === false || observedBits === 0) {
+    // Do not observe any updates.
```

Comment:
Note how both these checks let us do nothing at all, including more checks. It's also included in the other checks when we do have to do something so it's not an additional cost.

So by hoisting them we don't have to do any of the other checks except in the other branch.